ql/patterns/observable.hpp

```diff
@@ -89,18 +87,20 @@ namespace QuantLib {
         */
         void notifyObservers();
       private:
-        typedef boost::unordered_set<Observer*>::iterator iterator;
+        typedef std::unordered_set<Observer*>::iterator iterator;
         std::pair<iterator, bool> registerObserver(Observer*);
         Size unregisterObserver(Observer*);
-        boost::unordered_set<Observer*> observers_;
+        std::unordered_set<Observer*> observers_;
         ObservableSettings& settings_;
     };
 
     //! Object that gets notified when a given observable changes
     /*! \ingroup patterns */
     class Observer {
       public:
-        typedef boost::unordered_set<ext::shared_ptr<Observable> > set_type;
+        QL_DEPRECATED // TODO - make private and undeprecate
+        typedef std::unordered_set<ext::shared_ptr<Observable> > set_type;
+        QL_DEPRECATED // TODO - make private and undeprecate
         typedef set_type::iterator iterator;
```

Comment:
If we deprecate, we have to leave the deprecated `boost::unordered_set` for the time being and change it to `std::` when it moves to private.

Reply:
I've reverted back the publicly exposed typedefs to the boost version.

This pull rquest is still a draft and I may have a couple other changes to make in the next few days depending on the CI results.